src/clj/rems/application/commands.clj

```diff
@@ -417,14 +424,32 @@
            :application/member (:member cmd)
            :application/comment (:comment cmd)})))
 
+(defmethod command-handler :application.command/copy-as-new
+  [cmd application {:keys [create-application!]}]
+  (let [result (create-application! (:actor cmd) (map :catalogue-item/id (:application/resources application)))
+        _ (assert (:success result) {:result result})
+        new-app-id (:application-id result)]
+    (ok-with-data
+     {:application-id new-app-id}
+     {:event/type :application.event/draft-saved
+      :application/id new-app-id
+      :application/field-values (->> (:form/fields (:application/form application))
+                                     (map (fn [field]
+                                            [(:field/id field) (:field/value field)]))
+                                     (into {}))}
+     {:event/type :application.event/copied-from
```

Comment:
I find it a bit disturbing that `create-application!` emits the created event and then we emit other events here. Couldn't you emit the created event explicitly?

Reply:
I can try refactor it. `rems.db.applications/create-application!` needs some refactoring to do that.

Reply:
I'm fine with a TODO comment and refactoring in a follow-up PR

Reply:
👍